tests/queries/0_stateless/02843_backup_use_same_s3_credentials_for_base_backup.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+# Tag: no-fasttest - requires S3
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT -nm -q "
+    drop table if exists data;
+    create table data (key Int) engine=MergeTree() order by tuple();
+    insert into data select * from numbers(10);
+"
+
+function write_invalid_password_to_base_backup()
+{
+    local name=$1 && shift
+    local content
+    content=$($CLICKHOUSE_CLIENT -q "select replace(line, 'testtest', 'INVALID_PASSWORD') from s3($(s3_location $name/.backup), 'LineAsString')") || return 1
+    $CLICKHOUSE_CLIENT --param_content="$content" -q "insert into function s3($(s3_location $name/.backup), 'LineAsString') settings s3_truncate_on_insert=1 values ({content:String})"
+}
+
+function s3_location() { echo "'http://localhost:11111/test/backups/$CLICKHOUSE_DATABASE/use_same_s3_credentials_for_base_backup_base_$*', 'test', 'testtest'"; }
+
+$CLICKHOUSE_CLIENT --format Null -q "BACKUP TABLE data TO S3($(s3_location base))"
+$CLICKHOUSE_CLIENT --format Null -q "BACKUP TABLE data TO S3($(s3_location inc_1)) SETTINGS base_backup=S3($(s3_location base))"
+write_invalid_password_to_base_backup inc_1
+$CLICKHOUSE_CLIENT --format Null -q "BACKUP TABLE data TO S3($(s3_location inc_2)) SETTINGS base_backup=S3($(s3_location inc_1))" |& grep -m1 -o 'The request signature we calculated does not match the signature you provided. Check your key and signing method. (S3_ERROR)'
+$CLICKHOUSE_CLIENT --format Null -q "BACKUP TABLE data TO S3($(s3_location inc_3)) SETTINGS base_backup=S3($(s3_location inc_1)), use_same_s3_credentials_for_base_backup=1"
+
+$CLICKHOUSE_CLIENT --format Null -q "RESTORE TABLE data AS data FROM S3($(s3_location inc_1))" |& grep -m1 -o 'The request signature we calculated does not match the signature you provided. Check your key and signing method. (S3_ERROR)'
+$CLICKHOUSE_CLIENT --format Null -q "RESTORE TABLE data AS data_1 FROM S3($(s3_location inc_1)) SETTINGS use_same_s3_credentials_for_base_backup=1"
+$CLICKHOUSE_CLIENT --format Null -q "RESTORE TABLE data AS data_2 FROM S3($(s3_location inc_3)) SETTINGS use_same_s3_credentials_for_base_backup=1"
+
+exit 0
```

Comment:
What's the point of `exit 0` ?

Reply:
This is a common practice in bash:

```bash
function foo()
{
    local arg1=$1 && shift
    local arg1=$1 && shift
    ...
}
```

That way you do not rely on the order of args, i.e.

```bash
function foo()
{
    local arg1=$1
    local arg1=$2
    ...
}
```

Here it is not important, but is has zero harm, and later (if) more arguments will be added the diff will contain only one line of change

Reply:
By default bash inherit exit status of the last command, while this is not important for this test, since we have stderr and this is enough to ensure that everything is fine.

Hence explicit `exit 0`